core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/codesystem/version/CodeSystemVersionCreateRequest.java

```diff
@@ -117,6 +113,10 @@ public Boolean execute(ServiceProvider context) {
 				.map(codeSystemsByShortName::get)
 				.collect(Collectors.toList());
 		
+		if (codeSystemsToVersion.stream().anyMatch(cs -> cs.getUpgradeOf() != null)) {
+			throw new IllegalArgumentException("Upgrade codesystem can not be versioned.");
```

Comment:
Use `BadRequestException` for client errors. Also, include the offending Code System's ID in the message. Also, a test would be really nice to have 🙂.